plugins/woocommerce/src/Internal/DataStores/Orders/LegacyDataHandler.php

```diff
@@ -322,8 +322,12 @@ public function get_order_from_datastore( int $order_id, string $data_store_id =
 		};
 		$update_data_store_func->call( $order, $data_store );
 
-		// Read order.
+		// Read order (without triggering sync).
+		$prevent_sync_on_read = fn() => false;
+
+		add_filter( 'woocommerce_hpos_enable_sync_on_read', $prevent_sync_on_read, 999 );
 		$data_store->read( $order );
+		remove_filter( 'woocommerce_hpos_enable_sync_on_read', $prevent_sync_on_read );
```

Comment:
Maybe this instead for robustness:

```suggestion
		try {
			$data_store->read( $order );
		}
		finally {
			remove_filter( 'woocommerce_hpos_enable_sync_on_read', $prevent_sync_on_read );
		}
```

Reply:
Done in 186d2ac4adfdc17650b0cb11b802d656227ec193. Thanks @Konamiman! 🙇